### PR TITLE
NETOBSERV-1974: Refactor network events to return list of maps instead of strings

### PR DESCRIPTION
**Pull request by msherif1234**
## Description

Network events was orignally decode as string with this recent ovnk lib it will come as struct and the agent need to generate list of configmaps with those events
this effort to allow generating metrics with network events

## Dependencies

https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4818
## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [ ] Will this change affect NetObserv / Network Observability operator? If not, you can ignore the rest of this checklist.
* [ ] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

File: pkg/pbflow/proto.go [hunk 1]
```diff
@@ -99,13 +101,32 @@ func FlowToPB(fr *model.Record, s *ovnobserv.SampleDecoder) *Record {
...
+								Events: map[string]string{"Name": e.Name},
+							},
+							{
+								Events: map[string]string{"Namespace": e.Namespace},
+							},
+							{
+								Events: map[string]string{"Direction": e.Direction},
+							},
+						}
+						pbflowRecord.NetworkEventsMetadata = append(pbflowRecord.NetworkEventsMetadata, pbEvents...)
```
Comment (jotak) [inline pkg/pbflow/proto.go:129]: This shouldn't be one map per single event field, but one map per event (with all event fields in the same map)

```suggestion
						e := ovnmodel.ACLEvent{NetworkEvent: md}
						pbEvent := NetworkEvent{
								Events: map[string]string{
								    "Action": e.Action,
								    "Feature": e.Actor,
								    "Name": e.Name,
							            "Namespace": e.Namespace,
								    "Direction": e.Direction,
							},
						}
						pbflowRecord.NetworkEventsMetadata = append(pbflowRecord.NetworkEventsMetadata, pbEvents)
```

right?

File: pkg/pbflow/proto.go [hunk 2]
```diff
@@ -99,13 +101,32 @@ func FlowToPB(fr *model.Record, s *ovnobserv.SampleDecoder) *Record {
...
-		seen := make(map[string]bool)
+		seen := make(map[ovnmodel.NetworkEvent]bool)
+		pbflowRecord.NetworkEventsMetadata = make([]*NetworkEvent, 0)
 		for _, metadata := range fr.Metrics.NetworkEvents {
 			if !model.AllZerosMetaData(metadata) {
 				if md, err := s.DecodeCookie8Bytes(metadata); err == nil {
 					protoLog.Debugf("Network Events Metadata %v decoded Cookie: %v", metadata, md)
 					if !seen[md] {
-						pbflowRecord.NetworkEventsMetadata = append(pbflowRecord.NetworkEventsMetadata, md)
+						e := ovnmodel.ACLEvent{NetworkEvent: md}
```
Comment (jotak) [inline pkg/pbflow/proto.go:111]: Also you don't have the guarantee that md is of type NetworkEvent
Should be something like:
```go

					if acl, ok := md.(*ovnmodel.ACLEvent); ok {
						// (fill pbEvent)
					} else {
                                               // something generic here, such as:
                                               pbEvent := NetworkEvent{
								Events: map[string]string{
								    "Message": md.String()
                                                                },
                                                          },
					}
```

File: pkg/pbflow/proto.go (hunk 1 above)
Comment (msherif1234) [reply pkg/pbflow/proto.go]: yes I addressed this change

File: pkg/pbflow/proto.go (hunk 2 above)
Comment (msherif1234) [reply pkg/pbflow/proto.go]: I don't think we will hit a case that will generate string but can't get cast as ACLEvent but that change won't hurt

File: pkg/pbflow/proto.go [hunk 3]
```diff
@@ -100,13 +102,34 @@ func FlowToPB(fr *model.Record, s *ovnobserv.SampleDecoder) *Record {
...
+		pbflowRecord.NetworkEventsMetadata = make([]*NetworkEvent, 0)
 		for _, metadata := range fr.Metrics.NetworkEvents {
+			var pbEvent NetworkEvent
 			if !model.AllZerosMetaData(metadata) {
 				if md, err := s.DecodeCookie8Bytes(metadata); err == nil {
 					protoLog.Debugf("Network Events Metadata %v decoded Cookie: %v", metadata, md)
-					if !seen[md] {
-						pbflowRecord.NetworkEventsMetadata = append(pbflowRecord.NetworkEventsMetadata, md)
-						seen[md] = true
+					acl, ok := md.(*ovnmodel.ACLEvent)
```
Comment (jotak) [inline pkg/pbflow/proto.go:111]: md.String() is called three times below I guess it could be set in a var instead

**Review by jotak — APPROVED** [review]
LGTM

**Pull request merged**